File: max-consecutive-ones/max-consecutive-ones.scala
object Solution {
    def findMaxConsecutiveOnes(nums: Array[Int]): Int = {
    def countOnes(max: Int, sum: Int, i: Int): Int = {
      if (i == nums.length) math.max(sum, max)
      else {
        if (nums(i) == 0) countOnes(math.max(max, sum), 0, i + 1)
        else countOnes(max, sum + 1, i + 1)
      }
    }
    countOnes(0, 0, 0)
  }
}

File: max-consecutive-ones/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max = Integer.MIN_VALUE;
        int ones = 0;
        for (int i = 0; i < nums.length; i++){
            if (nums[i] != 0) ones = ones + 1;
            else {
                max = Math.max(max, ones);
                ones = 0;
            }
        }
        return Math.max(max, ones);
    }
}